app/src/main/java/com/afwsamples/testdpc/common/RestrictionManagerCompat.java

```diff
@@ -31,6 +31,12 @@
 public class RestrictionManagerCompat {
```

Comment:
Yes, you are right. I missed, that RestrictionEntry is Parcelable. In this case solution 1 looks good. 
I can implement it.